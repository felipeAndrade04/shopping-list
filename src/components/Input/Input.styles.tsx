import styled from 'styled-components/native';
import { InputProps } from './Input.types';
import { css } from 'styled-components';

export const InputContainer = styled.View`
  width: 100%;
`;

export const InputContent = styled.View<InputProps>`
  width: 100%;
  flex-direction: row;
  align-items: center;
  background-color: ${({ theme }) => theme.colors.white};
  border-radius: 6px;
  border: ${({ theme }) => `1px solid ${theme.colors.dark}`};
  opacity: ${(props) => (props.disabled ? 0.3 : 1)};
  transition: border-color 0.3s ease-in;

  ${({ theme, isFocused }) =>
    isFocused &&
    css`
      color: ${theme.colors.dark};
      border-color: ${theme.colors.main};
    `}

  ${({ theme, error }) =>
    error &&
    css`
      color: ${theme.colors.red};
      border-color: ${theme.colors.red};
    `}
`;

export const Input = styled.TextInput`
  height: 50px;
  flex: 1;
  color: ${({ theme }) => theme.colors.dark};
  font-size: 16px;
  padding: 8px 16px;
`;

export const InputTextError = styled.Text`
  color: ${({ theme }) => theme.colors.red};
  font-size: 12px;
  margin-top: 2px;
  margin-left: 4px;
`;
