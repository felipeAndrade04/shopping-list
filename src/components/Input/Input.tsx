import React, { useState } from 'react';
import * as S from './Input.styles';
import { InputProps } from './Input.types';
import { NativeSyntheticEvent, TextInputFocusEventData } from 'react-native';

export function Input({ disabled, ...props }: InputProps) {
  const [isFocused, setIsFocused] = useState(false);

  function onFocus() {
    setIsFocused(true);
  }

  function onBlur(e: NativeSyntheticEvent<TextInputFocusEventData>) {
    if (props.onBlur && e) {
      props.onBlur(e);
    }
    setIsFocused(false);
  }

  return (
    <S.Input
      {...props}
      onFocus={onFocus}
      onBlur={onBlur}
      isFocused={isFocused}
      editable={!disabled}
      disabled={disabled}
    />
  );
}
