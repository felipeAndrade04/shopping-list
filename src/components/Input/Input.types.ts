import { TextInputProps } from 'react-native';

export interface InputProps extends TextInputProps {
  disabled?: boolean;
  isFocused?: boolean;
  inputPassword?: boolean;
  error?: string;
  borderColor?: string;
  borderWidth?: number;
}
