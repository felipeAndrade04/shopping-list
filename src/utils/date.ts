const months = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];

export function formatDate(date: Date) {
  const day = date.getDay();
  const month = date.getMonth();

  return `${day} de ${months[month]}`;
}
