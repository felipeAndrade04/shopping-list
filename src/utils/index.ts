export * from './delay';
export * from './date';
