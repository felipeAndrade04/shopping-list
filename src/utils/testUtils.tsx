import React from 'react';
import { authReducer } from '@app/store';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import * as theme from '@app/theme';
import { configureStore } from '@reduxjs/toolkit';

let store = configureStore({
  reducer: {
    auth: authReducer,
  },
});

export const wrapper = ({ children }: { children: ReactNode }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>{children}</Provider>
  </ThemeProvider>
);

export function resetStore() {
  store = configureStore({
    reducer: {
      auth: authReducer,
    },
  });
}
