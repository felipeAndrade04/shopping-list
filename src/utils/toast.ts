import EventManager from './EvenManager';

export type ToastType = 'success' | 'info' | 'error';

export interface Toast {
  id?: string;
  type: ToastType;
  text: string;
  duration?: number;
}

export const toastEventManager = new EventManager();

export default function toast({ type, text, duration }: Toast) {
  toastEventManager.emit<Toast>('addtoast', {
    id: new Date().getTime().toString(),
    type,
    text,
    duration,
  });
}
