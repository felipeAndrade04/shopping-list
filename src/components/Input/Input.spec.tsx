import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { Input } from './Input';
import { wrapper } from '@app/utils/testUtils';

describe('<Input />', () => {
  it('Should be render correctly', () => {
    const tree = render(<Input />, { wrapper }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should change text', () => {
    let value = '';
    const onChangeText = (text: string) => (value = text);
    const { getByTestId } = render(<Input value={value} onChangeText={onChangeText} />, {
      wrapper,
    });

    const input = getByTestId('input');
    fireEvent.changeText(input, 'test');

    expect(value).toBe('test');
  });

  it('should show error message', () => {
    const { getByText } = render(<Input error="test error" />, {
      wrapper,
    });

    const textError = getByText('test error');

    expect(textError).toBeTruthy();
  });
});
