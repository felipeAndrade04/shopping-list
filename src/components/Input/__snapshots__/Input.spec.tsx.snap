// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Should be render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "width": "100%",
      },
    ]
  }
>
  <View
    isFocused={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#FFFFFF",
          "borderRadius": 6,
          "borderStyle": "solid",
          "borderWidth": 2,
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
          "transition": "border-color 0.3s ease-in",
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      editable={true}
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        Array [
          Object {
            "color": "#0D0A0B",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Poppins-Regular",
            "fontSize": 16,
            "height": 50,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
          },
        ]
      }
      testID="input"
    />
  </View>
</View>
`;
